### ISSUE TITLE
Some issues in Coding GRPO from Scratch: A Guide to Distributed Implementation with Qwen2.5-1.5B-Instruct

### ISSUE THREAD

**Issue by xuzhang37** [aburkov/theLMbook#4]
Very fantastic job! Just want to clarify some confusions in your GRPO implementation.
1. Why set `prompt_length` instead of `max_completion_length` for outputs?
![Image](https://github.com/user-attachments/assets/4009d1a3-c2a3-4ef2-8551-14b6efd8a80e)
2. In GRPO loss implementation, why use torch.exp for calculating ratio and estimate KL divergience? I just checked the original paper but did not find exp function in loss.
![Image](https://github.com/user-attachments/assets/35171469-7b41-44ae-bb62-6239e9601c19)
Here are the paper screenshots.
![Image](https://github.com/user-attachments/assets/ad1646a2-b7b2-4dca-8538-3c3ac541403e)
![Image](https://github.com/user-attachments/assets/0e58db0b-c770-4d80-89e9-b16cb9f9197d)

Looking forward to your reply and thanks for your great sharing!

**Comment by aburkov** [comment]
1. Because it's **starting at** `prompt_length` and not up to `prompt_length`
2. Its exp of a log, so one cancels the other

**Comment by xuzhang37** [comment]
Thanks for your reply. It is clear.
